tiledb/CMakeLists.txt

```diff
@@ -561,6 +624,76 @@ if (CURL_APPEND_LIBS)
     ${CURL_APPEND_LIBS})
 endif()
 
+############################################################
+# capnproto file generation
+############################################################
+
+if(TILEDB_SERIALIZATION)
+  if(TILEDB_CAPNP_EP_BUILT)
+    set(TILEDB_CAPNPROTOEXEPATH "${TILEDB_CAPNPEXEC_PATH}")
+  else()
+    find_package(CapnProto 0.6.1 #*not* EXACT since windows/*n*x using 0.8.0, 0.6.1 respectively
+      PATHS  ${TILEDB_EP_INSTALL_PREFIX} ${TILEDB_DEPS_NO_DEFAULT_PATH}
+    )
+    set(TILEDB_CAPNPROTOEXEPATH "${CAPNP_EXECUTABLE}")
+  endif()
+
+  #force (re-)gen to run each time so can alternate msvc/wsl builds without removing *-capnp.<h,c++>
+  if(WIN32)
+    add_custom_target(tiledb_genned_capnpfiles #ALL
+      COMMAND
+        powershell "../scripts/capnpupdate.ps1 "
```

Comment:
`${CMAKE_CURRENT_SOURCE_DIR}/../scripts`

Reply:
Generating into the build tree with capnp_generate_cpp is a reasonable idea, 
I am attempting that, but so far not succeeding, think it should be doable windows, 
hoping the mixed versions aren't an issue here (for *nix), 
I think there were at least some canpproto changes related to how that was implemented between 0.6.1 and 0.8.0.

Reply:
Sorry, two different issues here:
- if the scripts are kept, they must be invokved with `${CMAKE_CURRENT_SOURCE_DIR}/../scripts` so that the build works out of tree.
- regardless of the generation method (current scripts, cmake script, or macro), generating into the build tree will solve the problem you mentioned with two separate build trees for MSVC and WSL from the same source tree.

Reply:
Seems the capnproto variables CAPNPC_SRC_PREFIX AND CAPNPC_OUTPUT_DIR needed to allow gen'ing outside of project directory were added sometime after 0.8.0, not doc'd in 0.8.0 CapnProtoConfig.cmake.in and attempts to use them are ineffective,
so returning approach to capnpupdate.<ps1,sh> and modifying files as needed to generate into build tree (as the capnp[.exe] compiler itself does support specification of an output directory, had used this before experimentally, so know it does work in both 0.6.1 and 0.8.0)

Reply:
Seems the capnproto variables CAPNPC_SRC_PREFIX AND CAPNPC_OUTPUT_DIR needed to allow gen'ing outside of project directory were added sometime after 0.8.0, not doc'd in 0.8.0 CapnProtoConfig.cmake.in and attempts to use them are unavailable/ineffective,
so returning approach to capnpupdate.<ps1,sh> and modifying files as needed to generate into build tree (as the capnp[.exe] compiler itself does support specification of an output directory, had used this before experimentally, so know it does work in both 0.6.1 and 0.8.0)

Reply:
regarding the direct execution of commands
I had originally tried to do that, but was fighting cmake trying to get a correctly functioning command (IIRC problem had to do with setting the path so that capnp.exe could find its subordinate 'plugin(s)' as well as then invoking the command from within the same context) and realizing I had already coded functioning scripts (for dev use from command line) fell back to using them.